File: ios/BestChanceStudio/BestChanceStudio/Models/DimensionScore.swift
import Foundation
import SwiftData

@Model
final class DimensionScore {
    var session: ScoringSession?
    var dimensionId: String
    var score: Int
    var maxScore: Int

    init(dimensionId: String, score: Int, maxScore: Int) {
        self.dimensionId = dimensionId
        self.score = score
        self.maxScore = maxScore
    }
}

File: android/BestChanceStudio/app/src/main/kotlin/org/bestchancestudio/app/models/DimensionScore.kt
package org.bestchancestudio.app.models

import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.Index
import androidx.room.PrimaryKey

@Entity(
    tableName = "dimension_scores",
    foreignKeys = [
        ForeignKey(
            entity = ScoringSession::class,
            parentColumns = ["id"],
            childColumns = ["sessionId"],
            onDelete = ForeignKey.CASCADE
        )
    ],
    indices = [Index("sessionId")]
)
data class DimensionScore(
    @PrimaryKey(autoGenerate = true)
    val id: Long = 0,
    val sessionId: Long,
    val dimensionId: String,
    val score: Int,
    val maxScore: Int
)
